packages/react-reconciler/src/ReactFiberContext.js

```diff
@@ -113,19 +121,12 @@ function hasContextChanged(): boolean {
   return didPerformWorkStackCursor.current;
 }
 
-function isContextConsumer(fiber: Fiber): boolean {
-  return fiber.tag === ClassComponent && fiber.type.contextTypes != null;
-}
-
-function isContextProvider(fiber: Fiber): boolean {
-  return fiber.tag === ClassComponent && fiber.type.childContextTypes != null;
```

Comment:
The `tag` check was important here.

The code in `getUnmaskedContext` above says:

```js
  const hasOwnContext = isContextProvider(workInProgress);
  if (hasOwnContext) {
    // If the fiber is a context provider itself, when we read its context
    // we have already pushed its own child context on the stack. A context
    // provider should not "see" its own child context. Therefore we read the
    // previous (parent) context instead for a context provider.
    return previousContext;
  }
```

However, for factory components we used to execute `getUnmaskedContext` **before** pushing own context on the stack (because we don't know own context yet). In fact we called it twice: once before calling the factory, and once after we *know* it's a factory component.

The first call used to be ignored because `isContextProvider` was "unsure" when given `IndeterminateComponent` and returned `false`. However now `isContextProvider` is more confident when `IndeterminateComponent` is a context provider. Breaking the assumption that `getUnmaskedContext()` isn't called with a context provider until after it's been pushed.

Reply:
Fix in https://github.com/facebook/react/pull/13441